scripts/ensure_snap_builds.py

```diff
@@ -83,27 +83,50 @@ def ensure_lp_recipe(
     lp_owner = lp.people[LP_OWNER]
     lp_repo = lp.git_repositories.getDefaultRepository(target=lp_project)
     lp_ref = lp_repo.getRefByPath(path=flavor_branch)
+    lp_archive = lp.archives.getByReference(reference="ubuntu")
+    lp_snappy_series = lp.snappy_serieses.getByName(name="16")
     manifest = dict(
         auto_build=True,
+        auto_build_archive=lp_archive,
         auto_build_pocket="Updates",
         auto_build_channels={"snapcraft": "8.x/stable"},
         description=f"Recipe for {SNAP_NAME} {flavor_branch}",
         git_ref=lp_ref,
-        git_repository=lp_repo,
         information_type="Public",
         name=recipe_name,
         owner=lp_owner,
+        processors=[
+            "/+processors/amd64",
+            "/+processors/arm64",
+        ],
         store_channels=channels,
+        store_name=SNAP_NAME,
         store_upload=True,
+        store_series=lp_snappy_series,
     )
     try:
         recipe = lp.snaps.getByName(name=recipe_name, owner=lp_owner)
     except NotFound:
+        recipe = None
+
+    if not recipe:
         LOG.info(" Creating LP recipe %s", recipe_name)
-        (not dry_run) and lp.snaps.new(project=lp_project, **manifest)
-    else:
+        params = dict(**manifest)
+        params.pop("auto_build_channels")
+        recipe = (not dry_run) and lp.snaps.new(project=lp_project, **params)
+
+    if recipe:
         LOG.info(" Confirming LP recipe %s", recipe_name)
         updated = set()
+
+        recipe_processors = [
+            "/" + "/".join(p.self_link.split("/")[-2:]) for p in recipe.processors
+        ]
+        if (processors := manifest.pop("processors")) != recipe_processors:
+            updated |= {"processors"}
+            LOG.info("  Update processors: %s -> %s", recipe_processors, processors)
+            (not dry_run) and recipe.setProcessors(processors=processors)
```

Comment:
Snap recipes have to be set the processors with their own api. rather than just setting a list in the `recipes.processors`